### ISSUE TITLE
Redis DB keeps growing

### ISSUE THREAD

**Issue by oTree-org** [andrewgodwin/channels#315]
When I use my app, the number of `asgi` keys in my Redis DB keeps growing, and does not return to its baseline after use (or after restarting the server, etc). This sometimes results in an memory-over-limit error (>30 MB after 1 hour of use by about 100 people).

The keys look like this:

```
"asgi::chgroups:websocket.send!QnCLrDsVtZwK"
"asgi::chgroups:websocket.send!DzDyqqEiaaXB"
"asgi::chgroups:websocket.send!dBXxcujwEAcQ"
```

And their values correspond to my various channel-group names, e.g.:

```
{b'wait-page-26-page10-group2'}
{b'auto-advance-3fautb4x'}
{b'auto-advance-ky4g5ia0'}
{b'browser-bots-client-vrlwv5o6'}
{b'auto-advance-tqwtvyki'}
```

It seems most keys are getting deleted, but many are not. For example, I have a test suite that simulates clicking through ~500 pages. During the test run, the number of Redis keys jumps from 0 to 1000, and within a minute of completion it drops down to 300 and stays there. Most pages just have 1 websocket channel connection; it's the auto-advance-* groups, which account for 80% of the Redis keys.

Is it expected that the Redis DB accumulates ASGI keys over time that need to be flushed by my code explicitly?

Here is my [consumers.py](https://github.com/oTree-org/otree-core/blob/master/otree/channels/consumers.py); I am discarding each channel on disconnect.

Additional info:

- Same thing happens locally and on Heroku.
- It happens whether I use Chrome or Firefox.
- I'm using the latest Channels (channels==0.17.2) and Daphne (0.14.3), but this also happened with channels 0.13.
- Python 3.5

**Comment by andrewgodwin** [comment]
Ah, those are channel group tracking keys, which are set to expire after the group expiry time (one day), and moreover, it looks like the cleanup-on-group-removal code is not removing them ahead of that if a group discard is received.

Give me a little bit, I'll remove that tracking, as it's currently unused, hence the half-baked state.

**Comment by andrewgodwin** [comment]
This should now be fixed by `asgi_redis` version 0.14.1, which I just released. Sorry about missing those keys!

**Comment by oTree-org** [comment]
Looks fixed now. Thanks for the very quick fix! :100:
